### PR TITLE
refactor: Auth login using different token, add expired check

### PR DESCRIPTION
**Pull request by iwisunny**
Fix Home page search trigger two times componentDidMount

Refactor server routes

See: #467 #475

### CODE REVIEW

**Comment by iwisunny** [comment]
Removed all `noLogin` flag @whDongRui

File: src/pages/Home/index.jsx [hunk 1]
```diff
@@ -147,13 +143,13 @@ export default class Home extends Component {
...
     const { fixNav } = rootStore;
-    const { homeApps, isLoading, isProgressive } = appStore;
+    const { apps, isLoading, isProgressive } = appStore;
     const categories = categoryStore.categories;
-    const categoryId = match.params.category;
-    const appSearch = match.params.search;
-    const showApps = appSearch || Boolean(categoryId) ? homeApps.slice() : homeApps.slice(0, 3);
+
+    const { category, search } = match.params;
+    const showApps = category || search ? apps : apps.slice(0, 3);
```
Comment (whDongRui) [inline src/pages/Home/index.jsx]: The home page show newest apps  and category apps, add homeApps variable for storage  newest apps, becase query category apps will change apps data. If remove  homeApps variable, the newest apps show is error.

File: src/components/Layout/index.jsx [hunk 2]
```diff
@@ -71,12 +69,11 @@ export default class Layout extends React.Component {
...
       backBtn,
       hasSearch,
-      title,
-      noLogin
+      title
     } = this.props;
 
     const { isNormal, isDev, isAdmin } = this.props.user;
-    const hasMenu = (isDev || isAdmin) && !noLogin;
+    const hasMenu = isDev || isAdmin;
```
Comment (whDongRui) [inline src/components/Layout/index.jsx]: This noLogin is different parameter noLogin， this for judge home detail page has not sideNav . The previous PR noLogin rename isHome.

File: src/pages/Home/index.jsx (hunk 1 above)
Comment (iwisunny) [reply src/pages/Home/index.jsx]: Got, i will check it

File: src/components/Layout/index.jsx (hunk 2 above)
Comment (iwisunny) [reply src/components/Layout/index.jsx]: ok, so the name `noLogin` is not right, maybe `noSideNav` is better

**Pull request merged**